.jupyter/jupyterhub_config.py

```diff
@@ -232,6 +233,20 @@ def get_env(self):
 
     return env
 
+  async def get_url(self):
+    pods = oapi_client.resources.get(kind='Pod', api_version='v1')
+    pod = pods.get(name="jupyterhub-nb-%s" % escape(self.user.name), namespace=custom_notebook_namespace if custom_notebook_namespace else namespace)
+
+    url = None
+    try:
+        url = "http://%s:%d/user/%s/" % (pod["status"]["podIP"], pod['spec']['containers'][0]['ports'][0]['containerPort'], self.user.name)
+    except Exception as e:
+        self.log.error("Failed to produce pod url")
+        raise
```

Comment:
Will this also destroy the pod and unblock the user, or will they have to wait for the 10m timeout?

Reply:
IMO the exception would only happen if a) OpenShift is broken or b) the pod is  not there for some reason..at that point,  we are at a "standard" error situation, so the user might need to wait for the timeout, but that would be expected if any error happens. We are really only solving the specific situation here of JH pod dying during spawn...no need to change any other behaviours